network-mux/src/Network/Mux/Types.hs

```diff
@@ -126,8 +126,7 @@ type family HasResponder (mode :: MuxMode) :: Bool where
 --   serving downstream peers using server side of each protocol and getting
 --   updates from upstream peers using client side of each of the protocols.
 --
-newtype MiniProtocolBundle (mode :: MuxMode) =
-        MiniProtocolBundle [MiniProtocolInfo mode]
```

Comment:
I am not sure I am a fan of this change. newtypes > type synonyms. Why did you decided to do this?

Reply:
Perhaps there isn't other uses of `[MiniProtocolInfo mode]` to make it easy to make a mistake. But I also don't like type synonyms if that's the case. I much rather prefer the raw type

Reply:
It's unnecessary because we only have `[MiniProtocolInfo mode]` in one role. If we had two different roles for it, then a newtype(s) would make sense to differentiate them. Another argument is that its hiding what are we passing around (one needs one step more to find it out).  We could go one step forward and remove the type alias as well.

Reply:
Yes, I am in favor of transparency in this case.

Reply:
Fixed.